refactor: switch to mise

Vilsol opened the pull request:
Switch to mise as devbox seems to be more trouble than it's worth.

This also means that people should be able to easily have all the tooling needed to develop on Windows.

budak7273 committed:
feat: dev compose file that adds a pgadmin container

(cherry picked from commit 6a912b2a3ce3955891aaeb8b581b899c8f68ea8c)

Vilsol committed:
refactor: switch to mise
chore: update some tooling

Vilsol committed:
ci: install libwebp-dev dependency

Vilsol committed:
chore: upgrade a bunch of tools and dependencies

Vilsol committed:
chore: switch go tools to native tools

Vilsol committed:
chore: tidy after generation
fix: update to match new lru api

Vilsol committed:
chore: clean up old flakes

budak7273 reviewed (changes requested):
```
PS C:\Git\smr-api> mise run api
mise config files in C:\Git\smr-api are not trusted. Trust them? Yes
mise WARN  failed to resolve toolset: [C:\Git\smr-api\mise.toml] mc@2025-04-08T15-39-49Z: mc not found in mise tool registry
mise Installed executable into C:\Users\Rob\AppData\Local\mise\installs\go-junit-report\2.1.0\go-junit-report.exe
mise protoc-gen-go@1.36.6 ✓ installed
mise protoc@30.2 ✓ installed
mise go-junit-report@2.1.0 ✓ installed
mise go@1.24.2   ✓ installed
mise golangci-lint@2.0.2 ✓ installed
mise protoc-gen-go-grpc@1.5.1 ✓ installed                                                                               mise ERROR failed to install aqua:ariga/atlas@0.32.0
mise ERROR unsupported env: windows/amd64
mise ERROR Run with --verbose or MISE_VERBOSE=1 for more information
```

Vilsol committed:
refactor: switch to aqua for minio client

Vilsol committed:
feat: drop C dependency, use native webp

Vilsol committed:
ci: do not fail fast

Vilsol committed:
chore: tidy

budak7273 commented on .github/workflows/release.yml:
```diff
@@ -22,7 +22,7 @@ jobs:
       - name: Download dependencies
         run: |
-          sudo apt update && sudo apt install -y build-essential libpng-dev protobuf-compiler
+          sudo apt update && sudo apt install -y build-essential libpng-dev protobuf-compiler libwebp-dev
```
Is this package still needed now that we use a go native webp library? (16b33770dfed5eebb2699d8b02bd53332ed686d8)

budak7273 commented on .github/workflows/generated.yml:
```diff
@@ -28,12 +32,16 @@ jobs:
+      - uses: jdx/mise-action@v2
+
+      - name: Install dependencies
+        run: sudo apt -y install libwebp-dev
```
Is this package still needed now that we use a go native webp library? (16b33770dfed5eebb2699d8b02bd53332ed686d8)

budak7273 commented on .vscode/settings.json:
User-specific settings

Vilsol committed:
refactor: cleanup all libwebp references
chore: remove vscode settings

Vilsol committed:
refactor: remove libpng references

Vilsol committed:
docs: update README.md

Vilsol committed:
chore: make codecov ignore generated files

Vilsol committed:
chore: windows is special

Vilsol committed:
chore: always succeed

Vilsol committed:
chore: grafana now lives in 3001

budak7273 committed:
fix: do not auto start pak_parser on docker launch

budak7273 committed:
feat: use dev compose file
feat: switch to docker `--wait` to ensure containers start up without waiting longer than needed
fix: clarify message when minio account already exists

budak7273 committed:
feat: add pgadmin usage info
feat: add database migration troubleshooting info
chore: format readme

Vilsol committed:
ci: use env var to give github token

budak7273 reviewed (dismissed):
Works on my windows machine and on Vilsol's linux machine, and can connect to windows-hosted smr-frontend.

Vilsol committed:
chore: add back sleep, skip vt test if missing
